bpf/lib/common.h

```diff
@@ -664,6 +668,18 @@ enum metric_dir {
 #define DSR_IPV6_OPT_LEN	(sizeof(struct dsr_opt_v6) - 4)
 #define DSR_IPV6_EXT_LEN	((sizeof(struct dsr_opt_v6) - 8) / 8)
 
+/* Geneve option used to carry service addr and port for DSR.
+ *
+ * Class = 0x200 (unassigned according to [1])
+ * Type  = 0x20  (vendor-specific)
+ *
+ * [1]: https://www.iana.org/assignments/nvo3/nvo3.xhtml#geneve-option-class
+ */
+#define DSR_GENEVE_OPT_CLASS	0x200
+#define DSR_GENEVE_OPT_TYPE	0x20
+#define DSR_IPV4_GENEVE_OPT_LEN	2
+#define DSR_IPV6_GENEVE_OPT_LEN	5
```

Comment:
Can we declare these using the `sizeof` with target structures , then applying the multiplication to get the correct size? This would make the math easier to confirm it's correct. (It looks correct to me)

https://www.rfc-editor.org/rfc/rfc8926.html#name-tunnel-header-fields

Reply:
Is it something like this?

```
(sizeof(geneve_dsr_opt4) - sizeof(geneve_opt_hdr hdr)) / 4
(sizeof(geneve_dsr_opt6) - sizeof(geneve_opt_hdr hdr)) / 4
```